crates/jit-debug/src/gdb_jit_int.rs

```diff
@@ -2,9 +2,11 @@
 //! the __jit_debug_register_code() and __jit_debug_descriptor to register
 //! or unregister generated object images with debuggers.
 
-use std::pin::Pin;
-use std::ptr;
-use std::sync::Mutex;
+use alloc::boxed::Box;
+use alloc::vec::Vec;
+use core::pin::Pin;
+use core::ptr;
+use spin::mutex::Mutex;
```

Comment:
This unfortunately is a dependency we cannot use. Spinning on contention is almost never the right concurrency primitive as it's only particularly useful in a kernel setting when interrupts are disabled for example. I also unfortunately do not have a great answer of what to do as an alternative. The best I can think of is "panic on contention" in no_std mode but still use a real `std::sync::Mutex` in std mode.